File: solutions-c/set-2/09-pkcs7-padding.c
#include <stdio.h>
#include "../utils/padding.h"

int main()
{
  char input[20] = "YELLOW SUBMARINE";
  char expected_output[20] = "YELLOW SUBMARINE\x4\x4\x4\x4";
  int result = 0;
  pad_pkcs(input, 20, 16, 7, 20);
  for(unsigned int i = 0; i < 20; ++i)
  {
    printf("%c", input[i]);
    if(input[i] != expected_output[i])
    {
      result = -1;
      break;
    }
  }
  if(result == 0)
  {
    printf("\nSuccess!\n");
  }
}

File: solutions-python3/09-pkcs7-padding.py
#!/usr/bin/env python3

from libcryptopals.utils import pad_pkcs

input = "YELLOW SUBMARINE"
expected_output = "YELLOW SUBMARINE\x04\x04\x04\x04"
output = pad_pkcs(input, 7, 20)
print(output)
if output == expected_output:
  print('Success!')
